azext_iot/tests/test_iot_central_unit.py

```diff
@@ -48,103 +57,235 @@ def get_twin(self, device_id):
-        assert _get_aad_token(Cmd(), 'resource') == {
-            'accessToken': 'raw token 0 -b',
-            'expiresOn': 'value',
-            'subscription': 'raw token 1',
-            'tenant': 'raw token 2',
-            'tokenType': 'raw token 0 - A'
+        assert _get_aad_token(Cmd(), "resource") == {
+            "accessToken": "raw token 0 -b",
+            "expiresOn": "value",
+            "subscription": "raw token 1",
+            "tenant": "raw token 2",
+            "tokenType": "raw token 0 - A",
         }
 
-    def test_get_iot_hub_token_from_central_app_id(self, fixture_get_iot_central_tokens):
+    def test_get_iot_hub_token_from_central_app_id(
+        self, fixture_get_iot_central_tokens
+    ):
         from azext_iot.common._azure import get_iot_hub_token_from_central_app_id
 
         # Test to ensure get_iot_hub_token_from_central_app_id returns iothubTenantSasToken
-        assert get_iot_hub_token_from_central_app_id({}, 'app_id', 'api-uri') == 'iothubTenantSasToken'
+        assert (
+            get_iot_hub_token_from_central_app_id({}, "app_id", "api-uri")
+            == "iothubTenantSasToken"
+        )
 
 
-class TestDeviceTwinShow():
-    def test_device_twin_show_calls_get_twin(self, fixture_iot_token, fixture_bind_sdk, fixture_cmd):
-        result = subject.iot_central_device_show(fixture_cmd, device_id, app_id, 'api-uri')
+class TestDeviceTwinShow:
+    def test_device_twin_show_calls_get_twin(
+        self, fixture_iot_token, fixture_bind_sdk, fixture_cmd
+    ):
+        result = subject.iot_central_device_show(
+            fixture_cmd, device_id, app_id, "api-uri"
+        )
 
         # Ensure get_twin is called and result is returned
         assert result is device_twin_result
 
         # Ensure _bind_sdk is called with correct parameters
         assert fixture_bind_sdk.called is True
         args = fixture_bind_sdk.call_args
-        assert args[0] == ({'entity': resource}, SdkType.service_sdk)
+        assert args[0] == ({"entity": resource}, SdkType.service_sdk)
 
 
-@pytest.mark.skipif(not validate_min_python_version(3, 5, exit_on_fail=False), reason="minimum python version not satisfied")
-class TestMonitorEvents():
-    @pytest.mark.parametrize("timeout, exception", [
-        (-1, CLIError),
-    ])
+@pytest.mark.skipif(
+    not validate_min_python_version(3, 5, exit_on_fail=False),
+    reason="minimum python version not satisfied",
+)
+class TestMonitorEvents:
+    @pytest.mark.parametrize("timeout, exception", [(-1, CLIError)])
     def test_monitor_events_invalid_args(self, timeout, exception, fixture_cmd):
         with pytest.raises(exception):
             subject.iot_central_monitor_events(fixture_cmd, app_id, timeout=timeout)
+
+
+@pytest.mark.skipif(
+    not validate_min_python_version(3, 5, exit_on_fail=False),
+    reason="minimum python version not satisfied",
+)
+class TestEvents3Parser:
```

Comment:
I'm not thrilled with the parser tests living in the central unit test module since it's a generic IoT Hub message parser. But I suppose we can revisit this later.

Reply:
I'd be more than happy to move it for you, I figured it was a good idea to add it to either this or the IoTHub unit tests since they are more likely to get run than something just called event parser unit tests

Reply:
It would be great if you could move it to `test_iot_utility_unit.py`.